### PR TITLE
[4712] Site tools App

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4712

### CODE REVIEW

File: templates/web/site-config.ftl [hunk 1]
```diff
@@ -1,83 +1,46 @@
...
-~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-~ GNU General Public License for more details.
-~
-~ You should have received a copy of the GNU General Public License
-~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
--->
-
-<#if (envConfig.role! == 'admin' || envConfig.role! == 'developer')>
-<!doctype html>
-<html xmlns="http://www.w3.org/1999/xhtml">
```
Comment (rart) [inline templates/web/site-config.ftl]: I don't think we're ready to remove this. Rollback the permission changes.

File: ui/app/src/components/GraphiQL/GraphiQL.tsx [hunk 2]
```diff
@@ -80,18 +86,20 @@ function getGraphQLFetcher(url: string, method = 'post') {
 }
 
 function Graphi(props: GraphiQLProps) {
-  const [query, setQuery] = useState(window.localStorage.getItem(`${props.storageKey}graphiql:query`));
+  const { url, storageKey, method, embedded = false } = props;
+  const [query, setQuery] = useState(window.localStorage.getItem(`${storageKey}graphiql:query`));
```
Comment (rart) [inline ui/app/src/components/GraphiQL/GraphiQL.tsx]:
```suggestion
  const [query, setQuery] = useState(() => window.localStorage.getItem(`${storageKey}graphiql:query`));
```

**Review by rart — COMMENTED** [review]
(not done with full review)

File: ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx [hunk 3]
```diff
@@ -514,7 +519,7 @@ export default function SiteConfigurationManagement(props: SiteConfigurationMana
         <Box
           display="flex"
           flexGrow={1}
-          width={openDrawer ? `calc(100% - ${width}px )` : '100%'}
+          width="100%"
```
Comment (rart) [inline ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx]: Is this necessary. Isn't 100% the default? Specially with flex: 1

**Review by rart — CHANGES_REQUESTED** [review]

File: templates/web/site-config.ftl [hunk 4]
```diff
@@ -1,83 +1,47 @@
...
-~ but WITHOUT ANY WARRANTY; without even the implied warranty of
-~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-~ GNU General Public License for more details.
-~
-~ You should have received a copy of the GNU General Public License
-~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
--->
-
+<!--
+  ~ Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (rart) [inline templates/web/site-config.ftl]: 👆 outdated copy

File: templates/web/site-config.ftl [hunk 5]
```diff
@@ -1,83 +1,47 @@
...
-      text-align: center;
-      font-family: sans-serif;
-    }
-  </style>
-  <noscript>
-    <h1>Access Denied</h1>
-    <p>You don't have the necessary permissions to access this page.</p>
-  </noscript>
+  </body>
+  </html>
```
Comment (rart) [inline templates/web/site-config.ftl]: This should have been part of the permissions rollback above.

File: ui/app/scripts/deploy.js [hunk 6]
```diff
@@ -22,14 +24,23 @@ const //
...
-  jsNextScriptsFileContent = '<#include "/templates/web/common/js-global-context.ftl" />\n' + templateScripts;
+  indexContents = fs.readFileSync(`${PATH_BUILD}/index.html`).toString();
+
+let jsNextScriptsFileContent = '<#include "/templates/web/common/js-global-context.ftl" />\n';
+
+const parser = new htmlparser.Parser({
+  onopentag(name, attributes) {
+    if (name === 'script') {
+      jsNextScriptsFileContent += `<script src="${attributes.src}"></script>\n`;
+    } else if (name === 'link' && !attributes.rel.includes('icon')) {
```
Comment (rart) [inline ui/app/scripts/deploy.js]: We should be looking for all the rel=stylesheet. Why are you looking instead at whether it includes icon?

File: ui/app/src/utils/components.ts [hunk 7]
```diff
@@ -0,0 +1,176 @@
+/*
```
Comment (rart) [inline ui/app/src/utils/components.ts]: Rename to registerComponents

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/scripts/deploy.js (hunk 6 above)
Comment (aachavarria) [reply ui/app/scripts/deploy.js]: im excluding the icon one

**Review by rart — APPROVED** [review]

**Pull request merged**